packages/kit/src/runtime/server/page/load_data.js

```diff
@@ -189,6 +189,20 @@ export async function load_data({
 	return data;
 }
 
+/**
+ * @param {ArrayBuffer} buff
+ * @returns {string}
+ */
+function b64_encode(buff) {
+	if (globalThis.Buffer) {
+		return Buffer.from(buff).toString('base64');
+	}
+
+	// The Uint16Arrray(Uint8Array(..)) is done on purpose
+	// this way the code points are padded with 0's
+	return btoa(new TextDecoder('utf-16').decode(new Uint16Array(new Uint8Array(buff))));
+}
```

Comment:
in my own tests using kit's existing base64 worked well without needing all this. But maybe I was missing a use case.

https://github.com/sveltejs/kit/blob/master/packages/kit/src/runtime/server/page/crypto.js#L210-L239

Reply:
Node 20.6.1
In my testing kit's version is faster (~5x) for smaller sizes, at 1kib they are the same, at 32kib TextDecoder is 1.5x faster,
at 64kib TextDecoder is 2.5x and at 512kib is 10x.
for smaller sizes i don't think it matters if encoding a request takes 0.0006ms instead of 0.0001ms. But encoding 128kib is 0.5ms instead of 1.7ms